### PR TITLE
Make `ControlFlowBuilder.Clear` public.

### LINKED ISSUE #58765: [API Proposal]: Add `System.Reflection.Metadata.Ecma335.ControlFlowBuilder.Reset`.
### Background and motivation

One of the very few reference types in S.R.M, the `ControlFlowBuilder` class is an obstacle to low-allocation code, since it requires allocating one object of this type for each emitted method.

I propose to add a public method that resets `ControlFlowBuilder` objects, allowing consumers to reduce allocations by reusing them.

Such method already exists but it's internal and unused. It is named `Clear` but I think `Reset` is better since it's a more abstract word that more directly conveys what it does (and there are many other `Reset` methods with a similar purpose throughout the framework).

### API Proposal

```C#
namespace System.Reflection.Metadata.Ecma355
{
     public sealed class ControlFlowBuilder
     {
          public void Reset();
     }
}
```

### API Usage

```C#
BlobBuilder bb = /* ... */;
ControlFlowBuilder cfb = new ControlFlowBuilder();

EmitMethod1(bb, cfb);
cfb.Reset();
EmitMethod2(bb, cfb);
```

### Risks

No known risks.

### PR DESCRIPTION
**Pull request by teo-tsirpanis**
This PR makes the `System.Reflection.Metadata.Ecma335.ControlFlowBuilder.Clear()` method public, as just approved. I also added a test that reuses a `ControlFlowBuilder`.

Fixes #58765

### CODE REVIEW

File: src/libraries/System.Reflection.Metadata/tests/Metadata/Ecma335/Encoding/ControlFlowBuilderTests.cs [hunk 1]
```diff
@@ -418,5 +418,34 @@ public void Branch_LongInstruction_LongDistance()
...
+            static byte[] GenerateSampleIL(ControlFlowBuilder cfb)
+            {
+                var code = new BlobBuilder();
+                var il = new InstructionEncoder(code, cfb);
+
+                var label = il.DefineLabel();
+
+                il.MarkLabel(label);
+                il.OpCode(ILOpCode.Nop);
+                il.Branch(ILOpCode.Br_s, label);
```
Comment (buyaa-n) [inline src/libraries/System.Reflection.Metadata/tests/Metadata/Ecma335/Encoding/ControlFlowBuilderTests.cs:442]: Could add an `il` for `_lazyExceptionHandlers` too? To make sure it has always been cleared with the API

Comment (teo-tsirpanis) [reply src/libraries/System.Reflection.Metadata/tests/Metadata/Ecma335/Encoding/ControlFlowBuilderTests.cs]: Done.

**Review by buyaa-n — APPROVED** [review]
LGTM, thanks!

**Review by deeprobin — APPROVED** [review]
LGTM

**Pull request merged**